app/(app)/feed/detail/[entryId].tsx

```diff
@@ -1,38 +1,220 @@
+  footerContainer: {
+    position: 'absolute',
+    bottom: 0,
+    left: 0,
+    right: 0,
+    backgroundColor: theme.colors.gray1,
+    borderTopColor: theme.colors.gray4,
+    borderTopWidth: 1,
+    // fix animation
+    zIndex: 9999,
+  },
+  userAvatar: ({ index }: { index: number }) => ({
+    width: 28,
+    height: 28,
+    borderRadius: 14,
+    borderColor: theme.colors.gray10,
+    borderWidth: runtime.hairlineWidth,
+    backgroundColor: theme.colors.gray2,
+    alignItems: 'center',
+    justifyContent: 'center',
+    overflow: 'hidden',
+    transform: [{ translateX: -10 * index }],
+  }),
+  toolbarButton: ({ pressed }: { pressed: boolean }) => ({
+    padding: theme.spacing[1],
+    borderRadius: 6,
+    backgroundColor: pressed ? theme.colors.gray3 : theme.colors.gray4,
+  }),
+}))
+
+function EntryReadUsers({ users }: { users?: Array<Omit<User, 'emailVerified'>> }) {
+  const { styles } = useStyles(stylesheet)
+  const readUserAvatars = users?.map(i => i?.image).filter((i): i is string => i !== null) ?? []
+
+  return (
+    <AnimatedPressable
+      entering={FadeIn}
+      onPress={() => {
+        // TODO: open bottom sheet or modal of users which can navigate user profile route
+      }}
+    >
+      {readUserAvatars.length > 0 && (
+        <Animated.View style={{ flexDirection: 'row' }} entering={FadeIn}>
+          {readUserAvatars
+            .slice(0, READ_USER_AVATAR_COUNT)
+            .map((image, index) => (
+              <View key={image} style={styles.userAvatar({ index })}>
+                <Image
+                  style={{ height: '100%', width: '100%' }}
+                  source={{ uri: image }}
+                />
+              </View>
+            ))}
+          {readUserAvatars.length > 6 && (
+            <View style={styles.userAvatar({ index: READ_USER_AVATAR_COUNT })}>
+              <Text size={12}>+{readUserAvatars.length - READ_USER_AVATAR_COUNT}</Text>
+            </View>
+          )}
+        </Animated.View>
+      )}
+    </AnimatedPressable>
+  )
+}
+
+function EntryToolsbar({ entry }: { entry: Entry }) {
+  const { styles } = useStyles(stylesheet)
+  return (
+    <Row gap={8} align="center">
+      <Pressable
+        style={styles.toolbarButton}
+        onPress={() => {
+          Clipboard.setStringAsync(entry.url!).catch(console.error)
+        }}
+      >
+        <Iconify icon="mingcute:link-line" />
+      </Pressable>
+      <Pressable
+        style={styles.toolbarButton}
+        onPress={() => {
+          openExternalUrl(entry.url)
+            .catch(console.error)
+        }}
+      >
+        <Iconify icon="mingcute:world-2-line" />
+      </Pressable>
+      <Pressable
+        style={styles.toolbarButton}
+        onPress={async () => {
+          if (await Sharing.isAvailableAsync()) {
+            await Sharing.shareAsync(entry.url!)
+              .catch(console.error)
+          }
+        }}
+      >
+        <Iconify icon="mingcute:share-3-line" />
```

Comment:
[sharing icon In follow](https://github.com/RSSNext/Follow/blob/1edf5605e9621219e6cdfc732f12e6464113f765/icons/mgc/share_forward_cute_re.svg) is rightcopy so using origin ver